bin/kano-empty-trash

```diff
@@ -42,17 +43,44 @@ def empty_all_trash():
         # it thinks it must be root if sudo is in effect, but this is not the case here.
         os.system('sudo -H -u#{} SUDO_USER={} /usr/bin/kano-empty-trash '.format(uid, uname))
 
+def clean_old_dumps(dir, age):
+    '''
+    Cleans the core dump files which are set by the kernel.core_pattern
+
+    Args:
+        dir (str): This is where core dump files live.
+        age (int): List files older than age days old.
+    Raises:
+        OSError: Raised if the core dump dir isn there
 
+    TODO:
+        * Get the coredump_dir directly from an environment variable or through
+        the configuration file of kernel.core_pattern 
+    '''
+    if dir:
+        for file in os.listdir(dir):
+            full_path = os.path.join(dir, file)
+            now = datetime.now()
+            mtime = datetime.fromtimestamp(os.path.getmtime(file))
+            if (file.endswith('.dump.*') or file.endswith('.dump.gz')) and ((now - mtime) > timedelta(days=age)) and os.path.isfile(full_path):
+                try:
+                    os.remove(full_path)
+                except OSError, e:
+                    logger.error('Cannot delete:{} . {}'.format(e.filename, e.strerror))
+    else:
+        raise AttributeError
 
 if __name__ == '__main__':
-
     if os.getuid():
         if os.environ.has_key('HOME'):
             empty_user_trash(os.environ['HOME'])
         else:
             logger.error('HOME unset')
         sys.exit(0)
     else:
-        
         empty_all_trash()
+        try:
+            clean_old_dumps(COREDUMP_DIR, 5)
+        except AttributeError:
```

Comment:
Should we also be catching `OSError` for the case that the directory doesn't exist?

Reply:
Could do. The reason I haven't is that this directory is set already in `kernel.core_pattern`, so that's created by Debian by definition of the `core dump` process.